### PR TITLE
Update for new nightly

### PR DESCRIPTION
**Pull request by cramertj**

### CODE REVIEW

File: futures-core/src/future/mod.rs [hunk 1]
```diff
@@ -139,11 +139,11 @@ impl<'a, F: ?Sized + Future> Future for PinMut<'a, F> {
 if_std! {
     use std::boxed::{Box, PinBox};
 
-    impl<'a, F: ?Sized + Future> Future for Box<F> {
+    impl<'a, F: ?Sized + Future + Unpin> Future for Box<F> {
```
Comment (cramertj) [inline futures-core/src/future/mod.rs]: Note that I changed the `Box` impls not to project-- that allows us to be more conservative while we shake out whether or not we want `Box` to unconditionally be `Unpin`.

Comment (cramertj) [reply futures-core/src/future/mod.rs]: (personally I'm still not sold on that choice, but I don't think we should take a hard soundness dependency upon unresolved decisions like this)

File: futures-util/src/try_future/and_then.rs [hunk 2]
```diff
@@ -55,7 +55,7 @@ impl<A, B, F> Future for AndThen<A, B, F>
                 }
             };
 
-            // safe because we're using the `&mut` to do an assignment, not for moving out
+            // safe because we're using the `&mut` to do an setment, not for moving out
```
Comment (cramertj) [inline futures-util/src/try_future/and_then.rs]: Lol-- will fix

File: futures-channel/src/mpsc/mod.rs [hunk 3]
```diff
@@ -110,8 +110,8 @@ pub struct Sender<T> {
     maybe_parked: bool,
 }
 
-// Safe because we treat the `T` opaquely
-unsafe impl<T> Unpin for Sender<T> {}
+// We never project PInMut<Sender> to `PinMut<T>`
```
Comment (MajorBreakfast) [inline futures-channel/src/mpsc/mod.rs]: `PInMut` -> `PinMut`

File: futures-channel/src/mpsc/mod.rs [hunk 4]
```diff
@@ -137,7 +137,7 @@ pub struct Receiver<T> {
 pub struct UnboundedReceiver<T>(Receiver<T>);
 
 // Safe because we treat the `T` opaquely
```
Comment (aturon) [inline futures-channel/src/mpsc/mod.rs]: Should we update the comment?

File: futures-core/src/future/mod.rs (hunk 1 above)
Comment (aturon) [reply futures-core/src/future/mod.rs]: Good catch.

File: futures-channel/src/mpsc/mod.rs (hunk 4 above)
Comment (cramertj) [reply futures-channel/src/mpsc/mod.rs]: Done.

File: futures-channel/src/mpsc/mod.rs (hunk 3 above)
Comment (cramertj) [reply futures-channel/src/mpsc/mod.rs]: Done.

**Pull request merged**